prediction_market_agent_tooling/markets/omen/omen_graph_queries.py

```diff
@@ -0,0 +1,400 @@
+    return all_user_positions
+
+
+def get_market(market_id: str) -> OmenMarket:
+    market = requests.post(
+        OMEN_TRADES_SUBGRAPH,
+        json={
+            "query": _QUERY_GET_SINGLE_FIXED_PRODUCT_MARKET_MAKER,
+            "variables": {
+                "id": market_id,
+            },
+        },
+        headers={"Content-Type": "application/json"},
+    ).json()["data"]["fixedProductMarketMaker"]
+    return OmenMarket.model_validate(market)
+
+
+def construct_query_get_fixed_product_markets_makers(
+    include_creator: bool, filter_by: FilterBy
+) -> str:
+    query = """
+        query getFixedProductMarketMakers(
+            $first: Int!,
+            $outcomes: [String!],
+            $orderBy: String!,
+            $orderDirection: String!,
+            $creationTimestamp_gt: Int!,
+            $creator: Bytes = null,
+        ) {
+            fixedProductMarketMakers(
+                where: {
+                    isPendingArbitration: false,
+                    outcomes: $outcomes
+                    creationTimestamp_gt: $creationTimestamp_gt
+                    creator: $creator,
+                    answerFinalizedTimestamp: null
+                    resolutionTimestamp_not: null
+                },
+                orderBy: creationTimestamp,
+                orderDirection: desc,
+                first: $first
+            ) {
+                id
+                title
+                collateralVolume
+                usdVolume
+                collateralToken
+                outcomes
+                outcomeTokenAmounts
+                outcomeTokenMarginalPrices
+                fee
+                answerFinalizedTimestamp
+                resolutionTimestamp
+                currentAnswer
+                creationTimestamp
+                category
+                question {
+                    id
+                    answerFinalizedTimestamp
+                    currentAnswer                                  
+                }
+                condition {
+                    id
+                    outcomeSlotCount
+                }
+            }
+        }
+    """
+
+    if filter_by == FilterBy.OPEN:
+        query = query.replace("resolutionTimestamp_not: null", "")
+    elif filter_by == FilterBy.RESOLVED:
+        query = query.replace("answerFinalizedTimestamp: null", "")
+    elif filter_by == FilterBy.NONE:
+        query = query.replace("answerFinalizedTimestamp: null", "")
+        query = query.replace("resolutionTimestamp_not: null", "")
+    else:
+        raise ValueError(f"Unknown filter_by: {filter_by}")
+
+    if not include_creator:
+        # If we aren't filtering by query, we need to remove it from where, otherwise "creator: null" will return 0 results.
+        query = query.replace("creator: $creator,", "")
+
+    return query
+
+
+def get_omen_markets(
+    outcomes: list[str],
+    sort_by: SortBy,
+    filter_by: FilterBy,
+    first: int = 1000,  # max limit for markets
```

Comment:
Let's not default to 1000; both `get_{manifold, polymarket}_markets` don't have a default value for limit, so let's keep it consistent, please. No need for this, or is there?

```suggestion
    first: int
```

Reply:
Also, if we change order of arguments, it will create breaking changes for the repositories who use PMAT as a dependency, although it's unnecessary. Can you change it back to 

```
def get_omen_markets(
    first: int,
    outcomes: list[str],
    sort_by: SortBy,
    filter_by: FilterBy,
    created_after: t.Optional[datetime] = None,
    creator: t.Optional[HexAddress] = None,
    excluded_questions: set[str] | None = None,
) -> list[OmenMarket]:
```

please

Reply:
Changed back. Reason was that `first` had a default option, not needed anymore.